psa-crypto/src/types/status.rs

```diff
@@ -162,6 +162,39 @@ impl From<Status> for psa_crypto_sys::psa_status_t {
     }
 }
 
+#[cfg(feature = "with-mbed-crypto")]
+impl From<Error> for psa_crypto_sys::psa_status_t {
```

Comment:
If you have this then you don't need to have the whole match again above, in the `From<Status> for psa_status_t`, you can just do `Status::Error(error) => error.into()`

Reply:
Good point! Will push this modification.

Reply:
Thanks! This was actually one of the things that I needed for the `ResponseStatus` update :)